packages/eas-cli/src/utils/__tests__/profiles-test.ts

```diff
@@ -98,6 +104,53 @@ describe(getProfilesAsync, () => {
     expect(getBuildProfileAsync).toBeCalledWith(easJsonAccessor, Platform.IOS, 'custom-profile');
   });
 
+  describe('node version', () => {
+    const nodeVersion = '14.17.1';
+
+    it('is read from profile', async () => {
+      const easJsonAccessor = EasJsonAccessor.fromProjectPath(projectDir);
+      getBuildProfileAsync.mockImplementation(async () => {
+        return { node: nodeVersion } as BuildProfile<Platform.ANDROID>;
+      });
+      const result = await getProfilesAsync({
+        easJsonAccessor,
+        platforms: [Platform.ANDROID],
+        profileName: 'custom-profile',
+        type: 'build',
+        projectDir,
+      });
+
+      expect(result[0].profile.node).toBe(nodeVersion);
+    });
+
+    describe('with .nvmrc', () => {
+      beforeEach(async () => {
+        await fs.writeFile(path.join(projectDir, '.nvmrc'), nodeVersion);
```

Comment:
Is it using some kind of mock filesystem or temporary dir?

Reply:
it writes the file to fixtures directory

Reply:
Ok, so I think it would be good to mock `fs`. We already have a mock for it using `fs` from `memfs` in this repo.

Here's how you can do this using this mock (just do `jest.mock('fs')`) and `vol` from the `memfs` package https://github.com/expo/eas-cli/blob/fd99365655872dc6d7ce33e762c9bd3a99e0696a/packages/eas-cli/src/project/ios/__tests__/bundleIdentifier-test.ts